cf/post-register-lambda-v2.yaml

```diff
@@ -0,0 +1,229 @@
+
+            const convertToDynamoDbRecord = (data) => AWS.DynamoDB.Converter.marshall(
+              data, { convertEmptyValues: false },
+            );
+
+            const db = new AWS.DynamoDB({
+              region,
+            });
+
+            const params = {
+              TableName: inviteAccessTableName,
+              FilterExpression: '(userEmail = :userEmail)',
+              ExpressionAttributeValues: {
+                ...convertToDynamoDbRecord({ ':userEmail': userEmail }),
+              },
+            };
+            const response = await db.scan(params).promise();
+            if (!response.Items.length) {
+              return;
+            }
+
+            // add those rows in user-access
+            const experimentIds = [];
+            await Promise.all(response.Items.map((item) => {
+              const projectId = item.projectId.S;
+              const experimentId = item.experimentId.S;
+              experimentIds.push(experimentId);
+              const role = item.role.S;
+
+              const addRoleParams = {
+                TableName: userAccessTableName,
+                Item: {
+                  userId,
+                  experimentId,
+                  projectId,
+                  role,
+                  createdDate: new Date().toISOString(),
+                },
+              };
+
+              const docClient = new AWS.DynamoDB.DocumentClient({
+                region,
+              });
+
+              return docClient.put(addRoleParams).promise();
+            }));
+
+            // remove rows from invite-access
+            await Promise.all(experimentIds.map((id) => {
+              const key = convertToDynamoDbRecord({
+                experimentId: id,
+                userEmail,
+              });
+
+              const dynamoParams = {
+                TableName: inviteAccessTableName,
+                Key: key,
+              };
+
+              return db.deleteItem(dynamoParams).send();
+            }));
+          };
+
+          const addUserPermissionV2 = async (event) => {
+            const {
+              email,
+              sub,
+            } = event.request.userAttributes;
+
+            const sns = new AWS.SNS({ region });
+
+            const message = {
+              userId: sub,
+              userEmail: email,
+            };
+
+            await sns.publish({
+              Message: JSON.stringify(message),
+              TopicArn: process.env.TOPIC_ARN,
+              MessageAttributes: {
+                type: {
+                  DataType: 'String',
+                  StringValue: 'PostUserRegistration'
+                }
+              }
+            }, () => {}).promise();
+          };
+
+          exports.handler = async (event, context, callback) => {
+            await addUserPermissionV1(event);
+            await addUserPermissionV2(event);
+
+            callback(null, event);
+          };
+
+  LoggingGroup:
```

Comment:
I think this logging group might not be doing anything. Can you check?

Reply:
This is because the Lambda never throws an error and we don't log anything. I'll add some logs in the lambda function to show some logs.